unit-tests/shared/src/test/scala/org/scalanative/testsuite/javalib/net/DatagramSocketTest.scala

```diff
@@ -0,0 +1,308 @@
+    try {
+      assertTrue("Created socket with incorrect port", ds.getLocalPort() != 0)
+      assertEquals(
+        "Created socket with incorrect address",
+        InetAddress.getLocalHost(),
+        ds.getLocalAddress()
+      )
+    } finally {
+      ds.close()
+    }
+  }
+
+  @Test def close(): Unit = {
+    val ds = new DatagramSocket(0)
+    ds.close()
+    val dp = new DatagramPacket(
+      "Test String".getBytes,
+      11,
+      InetAddress.getLocalHost,
+      0
+    )
+    assertThrows(classOf[IOException], ds.send(dp))
+  }
+
+  @Test def getLocalPort(): Unit = {
+    val ds = new DatagramSocket()
+    try {
+      assertTrue("Returned incorrect port", ds.getLocalPort != 0)
+    } finally {
+      ds.close()
+    }
+  }
+
+  @Test def getPort(): Unit = {
+    val ds = new DatagramSocket()
+    try {
+      assertEquals(
+        "Expected -1 for remote port as not connected",
+        -1,
+        ds.getPort()
+      );
+      val port = 49152 // any valid port, even if it is unreachable
+      ds.connect(InetAddress.getLocalHost(), port)
+      assertEquals("getPort returned wrong value", port, ds.getPort())
+    } finally {
+      ds.close()
+    }
+  }
+
+  @Test def receiveBufferSize(): Unit = {
+    // This test basically checks that getReceiveBufferSize &
+    // setReceiveBufferSize do not unexpectedly throw and that the former
+    // returns a minimally sane value.
+    //
+    // The Java 8 documentation at URL
+    // https://docs.oracle.com/javase/8/docs/api/java/net/\
+    //     Socket.html#setReceiveBufferSize-int- [sic trailing dash]
+    // describes the argument for setReceiveBufferSize(int) &
+    // setSendBufferSize(int) as a _hint_ to the operating system, _not_
+    // a requirement or demand.  This description is basically unaltered
+    // in Java 10.
+    //
+    // There are a number of reasons the operating system can choose to
+    // ignore the hint. Changing the buffer size, even before a bind() call,
+    // may not be implemented. The buffer size may already be at its
+    // maximum.
+    //
+    // Since, by definition, the OS can ignore the hint, it makes no
+    // sense to set the size, then re-read it and see if it changed.
+    //
+    // The sendBuffersize test refers to this comment.
+    // Please keep both tests synchronized.
+
+    val ds = new DatagramSocket(null)
+    try {
+      val prevValue = ds.getReceiveBufferSize
+      assertTrue(prevValue > 0)
+      ds.setReceiveBufferSize(prevValue + 100)
+    } finally {
+      ds.close()
+    }
+  }
+
+  @Test def sendBufferSize(): Unit = {
+    // This test basically checks that getSendBufferSize &
+    // setSendBufferSize do not unexpectedly throw and that the former
+    // returns a minimally sane value.
+    // See more extensive comments in setBufferSize test.
+
+    val ds = new DatagramSocket(null)
+    try {
+      val prevValue = ds.getSendBufferSize
+      assertTrue(prevValue > 0)
```

Comment:
I suspect this code was copied from existing unit-test code.
If so, at the time that code was written one or more of the
following were probably true:
1) Scala Native and or JUnit did not have an "assertTrue(msg, cond)" implementation.
2) The original author (most probably me, so I can slag) was either lazy or over-worked and did not use the idiom)

Here, in `recvBufferSize`, and probably a few other places,
something like (raw, not compiled): 
`assertTrue(s"previousValue ${prevValue} <= 0". Unlike
`assertEquals`, `assertTrue` does not print out its arguments.

Having the arguments printed in the message may feel
picky but it saves much time and aggravation when one
is trying to figure out why something impossible happened and one's feet are in the fire, wishing for the traditional
gunpowder nosegay.